src/sentry/api/endpoints/integrations/sentry_apps/installation/index.py

```diff
@@ -43,9 +44,17 @@ def post(self, request: Request, organization) -> Response:
         if not serializer.is_valid():
             return Response(serializer.errors, status=400)
 
-        # check for an exiting installation and return that if it exists
         slug = serializer.validated_data.get("slug")
+
+        # only published or owned apps are allowed to be installed
+        app = SentryApp.objects.filter(slug=slug).first()
+        if not app:
+            return Response(status=404)
+        if not app.status == SentryAppStatus.PUBLISHED and app.owner_id != organization.id:
```

Comment:
Should this be an OR?

Reply:
Here is the truth table. We allow the app setup if it is published or if it belongs to the same org. Negating, we disallow only if it is not published AND does not belong to the same org.
```
|                                         | app.owner_id == organization.id | app.owner_id != organization.id |
|-----------------------------------------|---------------------------------|---------------------------------|
| app.status == SentryAppStatus.PUBLISHED | False (pass)                    | False (pass)                    |
| app.status != SentryAppStatus.PUBLISHED | False (pass)                    | True (return 404)               |
```

Reply:
Right, that makes sense!